x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/ingest/GeoGridProcessor.java

```diff
@@ -0,0 +1,370 @@
+import org.elasticsearch.h3.H3;
+import org.elasticsearch.ingest.AbstractProcessor;
+import org.elasticsearch.ingest.ConfigurationUtils;
+import org.elasticsearch.ingest.IngestDocument;
+import org.elasticsearch.ingest.Processor;
+import org.elasticsearch.search.aggregations.bucket.geogrid.GeoTileUtils;
+import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.spatial.common.H3CartesianUtil;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.elasticsearch.common.geo.GeometryParserFormat.GEOJSON;
+import static org.elasticsearch.common.geo.GeometryParserFormat.WKT;
+
+/**
+ *  The geo-grid-processor converts a geohash, geotile or geohex tile cell definition into a valid regular polygon approximating the tile.
+ */
+public final class GeoGridProcessor extends AbstractProcessor {
+    public static final String TYPE = "geo_grid";
+
+    private final FieldConfig config;
+    private final boolean ignoreMissing;
+    private final GeometryParserFormat targetFormat;
+    private final TileFieldType tileFieldType;
+
+    GeoGridProcessor(
+        String tag,
+        String description,
+        FieldConfig config,
+        boolean ignoreMissing,
+        GeometryParserFormat targetFormat,
+        TileFieldType tileFieldType
+    ) {
+        super(tag, description);
+        this.config = config;
+        this.ignoreMissing = ignoreMissing;
+        this.targetFormat = targetFormat;
+        this.tileFieldType = tileFieldType;
+    }
+
+    public static class FieldConfig {
+        private final String field;
+        private final String targetField;
+        private final String parentField;
+        private final String childrenField;
+        private final String nonChildrenField;
+        private final String depthField;
+
+        FieldConfig(
+            String field,
+            String targetField,
+            String parentField,
+            String childrenField,
+            String nonChildrenField,
+            String depthField
+        ) {
+            this.field = field;
+            this.targetField = targetField;
+            this.parentField = parentField;
+            this.childrenField = childrenField;
+            this.nonChildrenField = nonChildrenField;
+            this.depthField = depthField;
+        }
+
+        public String field(String key) {
+            return switch (key) {
+                case "field" -> field;
+                case "target_field" -> targetField;
+                case "parent_field" -> parentField;
+                case "children_field" -> childrenField;
+                case "non_children_field" -> nonChildrenField;
+                case "depth_field" -> depthField;
```

Comment:
I think depth should be cjange to precision that is how we name it in the grid aggregations.